server/index.js

```diff
@@ -0,0 +1,61 @@
+const express = require("express");
+const app = express();
+const port = 5050;
+
+app.use(express.urlencoded({ extended: false }));
+app.use(express.json());
+
+let todos = [
+  {
+    id: 1,
+    description: "eat",
+    completed: "false",
+  },
+  {
+    id: 2,
+    description: "sleap",
+    completed: "true",
+  },
+  {
+    id: 3,
+    description: "work",
+    completed: "false",
+  },
+];
+
+app.get("/", (request, response) => response.status(200).json(todos));
+
+app.post("/todo", (request, response) => {
+  const newTodo = {
+    id: todos.length + 1,
+    description: request.body.description,
+    completed: request.body.completed,
+  };
+  todos.push(newTodo);
+  response.status(201).json(newTodo);
+});
+
+app.put("/todo/:id", (request, response) => {
+  const id = parseInt(request.params.id);
+  const updatedTodo = todos.find((t) => t.id === id);
+  if (!updatedTodo) {
+    return response.status(404).json({ error: "Todo not found" });
+  }
+  updatedTodo.description = request.body.description || updatedTodo.description;
+  updatedTodo.completed = request.body.completed || updatedTodo.completed;
```

Comment:
its working now. but is should change it?

Reply:
Yep, it needs to be changed to prevent the following bug to happen:

Example bug scenario:
The task is already completed but the Request.body (i.e. the user) says it wants to mark it as "not completed" again. 

In object language, something like the following is happening:
```ts
request.body = { description: 'test2', completed: false };

// before
updatedTodo = { id: 1, description: 'test', completed: true };

// ... the update code ...

// after
updatedTodo = { id: 1, description: 'test2', completed: true };
```

It should be resulting in:

```ts
updatedTodo = { id: 1, description: 'test2', completed: false };
```

Reply:
but then i should not update completed? but always return false?

Reply:
It needs to be updated. But it needs to:

"UN"complete (false) when it is already completed (true)
Complete (true) when it is uncompleted (false)


The expression `false || true` will always return true. Which is wrong. The user wants to set it to "false".